gms/gossiper.cc

```diff
@@ -758,9 +758,10 @@ future<> gossiper::update_live_endpoints_version() {
 }
 
 future<std::set<inet_address>> gossiper::get_live_members_synchronized() {
-    auto live_members = gossiper::get_live_members();
-    co_await replicate_live_endpoints_on_change();
-    co_return live_members;
+    co_await container().invoke_on(0, [] (gms::gossiper& g) {
+        return g.replicate_live_endpoints_on_change();
+    });
+    co_return gossiper::get_live_members();
```

Comment:
The purpose of https://github.com/scylladb/scylladb/pull/12540 was to ensure that the value returned by this API was observed by all other shards.

This change removes the guarantee, so it undoes the goal that https://github.com/scylladb/scylladb/pull/12540 was supposed to achieve while leaving in place all its negative side effects.